### PR TITLE
VACMS-6352: Updates Drupal to 9.3.6

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Relates to #6352.  There's already a PR for this, but I think I b0rked it so I'm opening this to see where I went wrong 😭 

## QA steps

As user _uid_ with _user_role_
1. Do this
2. Then that
3. Then validate Acceptance Criteria from issue
- [ ] This
- [ ] That
- [ ] The other thing

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide CMS Team`
  - [ ] `⭐️ Content ops`
  - [ ] `⭐️ CMS Experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product Support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/modules/custom/va_gov_db/va_gov_db.install [hunk 1]
```diff
@@ -110,7 +110,10 @@ function _va_gov_rename_vocab_term($oldname, $newname, $vocabulary_name = NULL)
     ':oldname' => $oldname,
     ':newname' => $newname,
   ];
-  $terms = taxonomy_term_load_multiple_by_name($oldname, $vocabulary_name);
+  $terms = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->loadByProperties([
+    'name' => $oldname,
+    'vid' => $vocabulary_name,
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_db/va_gov_db.install]: Can wait and test this later the next time we need it, we'll know if it worked

File: docroot/modules/custom/va_gov_graphql/va_gov_graphql.module [hunk 2]
```diff
@@ -26,3 +27,35 @@ function va_gov_graphql_theme($existing, $type, $theme, $path) {
     ],
   ];
 }
+
+/**
+ * Get a list of menus excluding Drupal-defined menus.
+ *
+ * This is a replacement for menu_ui_get_menus(), which was deprecated in
+ * Drupal 9.3.0.
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_graphql/va_gov_graphql.module]: Needs an @return

File: docroot/modules/custom/va_gov_migrate/src/EventSubscriber/MessageLogger.php [hunk 3]
```diff
@@ -61,8 +69,16 @@ class MessageLogger implements EventSubscriberInterface {
   /**
    * {@inheritdoc}
    */
-  public function __construct(StateInterface $state) {
+  public function __construct(StateInterface $state, FileUrlGeneratorInterface $fileUrlGenerator) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_migrate/src/EventSubscriber/MessageLogger.php]: None of our currently active migrations are using this.   So not critical for testing.

File: tests/phpunit/Security/RolesPermissionsTest.php [hunk 4]
```diff
@@ -539,210 +517,198 @@ public function expectedPerms() {
...
+          'view vba_facility revisions',
+          'view vet_center revisions',
+          'view video revisions',
+          'view workbench access information',
+        ],
       ],
       [
         'admnistrator_users',
         [
-          'access shortcuts',
```
Comment (swirtSJW) [inline tests/phpunit/Security/RolesPermissionsTest.php]: I am not following why these are removed.   Is there some context for this?

Comment (ndouglas) [reply tests/phpunit/Security/RolesPermissionsTest.php]: yep, https://www.drupal.org/node/3193348

Comment (ndouglas) [reply tests/phpunit/Security/RolesPermissionsTest.php]: >One exception has been that uninstalled modules did not remove the permissions they defined from user roles. Before Drupal 9.3.0, a site administrator could uninstall a module, then install it again, and the permissions previously assigned to user roles would be in effect. Starting with Drupal 9.3.0, the administrator must configure permissions again when re-installing the module.

>An update function is provided to remove invalid permissions from existing roles. The update function logs the permissions removed from each role.

File: tests/phpunit/Security/RolesPermissionsTest.php [hunk 5]
```diff
@@ -539,210 +517,198 @@ public function expectedPerms() {
...
-        'view unpublished paragraphs',
-        'view va_form revisions',
-        'view vamc_operating_status_and_alerts revisions',
-        'view vamc_system_policies_page revisions',
-        'view vba_facility revisions',
-        'view vet_center revisions',
-        'view video revisions',
-        'view workbench access information',
-      ],
+        [
```
Comment (swirtSJW) [inline tests/phpunit/Security/RolesPermissionsTest.php]: Seems like the indention changed and is making it look like these changed.

Comment (ndouglas) [reply tests/phpunit/Security/RolesPermissionsTest.php]: yeah, the indentation was incorrect before.

File: tests/phpunit/Security/RolesPermissionsTest.php [hunk 6]
```diff
@@ -70,11 +69,8 @@ public function expectedPerms() {
           'access mobile_vet_centers entity browser pages',
           'access promo_blocks_browser entity browser pages',
           'access q_a_browser entity browser pages',
-          'access site-wide contact form',
           'access toolbar',
-          'access vet_centers entity browser pages',
           'add user_history entities',
-          'create terms in audience_tags',
```
Comment (swirtSJW) [inline tests/phpunit/Security/RolesPermissionsTest.php]: I 'm not following why all these are changing.   Did they go away in 9.3?

Comment (ndouglas) [reply tests/phpunit/Security/RolesPermissionsTest.php]: It just means that Drupal no longer has any knowledge of that permission -- it doesn't actually exist, so Drupal removed it from the user's permissions.  For instance, `access site-wide contact form` is defined in the Contact module, which we don't have enabled.

File: tests/phpunit/Security/RolesPermissionsTest.php [hunk 7]
```diff
@@ -94,11 +90,11 @@ public function expectedPerms() {
         [
           'access administration pages',
           'access files overview',
-          'access image_browser entity browser pages',
           'access media overview',
-          'access media_browser entity browser pages',
           'access taxonomy overview',
           'access user profiles',
+          'administer cms custom menu access',
```
Comment (swirtSJW) [inline tests/phpunit/Security/RolesPermissionsTest.php]: This was just taken away from content admin recently.  It should not be coming back.

File: tests/phpunit/Security/RolesPermissionsTest.php [hunk 8]
```diff
@@ -765,7 +730,6 @@ public function expectedPerms() {
           'manage password reset',
           'use text format rich_text',
           'use text format rich_text_limited',
-          'view cms_help_center in toolbar',
```
Comment (swirtSJW) [inline tests/phpunit/Security/RolesPermissionsTest.php]: Why is this one going?

File: tests/phpunit/Security/RolesPermissionsTest.php (hunk 7 above)
Comment (ndouglas) [reply tests/phpunit/Security/RolesPermissionsTest.php]: Ya, I see it removed [here](https://github.com/department-of-veterans-affairs/va.gov-cms/commit/d4c001644d7f212a31e0dea561b17a3fb5241c22).  I'll remove it.

**Comment by cweagans** [comment]
@ndouglas FYI @swirtSJW had a few review comments in https://github.com/department-of-veterans-affairs/va.gov-cms/pull/7917 -- I closed that PR in favor of this one.

**Comment by ndouglas** [comment]
Thanks @cweagans, I think that's all taken into account 😃

File: config/sync/user.role.authenticated.yml [hunk 9]
```diff
@@ -22,11 +33,8 @@ permissions:
   - 'access mobile_vet_centers entity browser pages'
   - 'access promo_blocks_browser entity browser pages'
   - 'access q_a_browser entity browser pages'
-  - 'access site-wide contact form'
   - 'access toolbar'
-  - 'access vet_centers entity browser pages'
   - 'add user_history entities'
-  - 'create terms in audience_tags'
```
Comment (swirtSJW) [inline config/sync/user.role.authenticated.yml]: Seems like an odd one to be losing.

File: config/sync/user.role.content_admin.yml [hunk 10]
```diff
@@ -90,8 +123,6 @@ permissions:
   - 'use editorial transition create_new_draft'
   - 'use editorial transition publish'
   - 'use editorial transition review'
-  - 'use editorial transition stage_for_publishing'
-  - 'use moderation sidebar'
```
Comment (swirtSJW) [inline config/sync/user.role.content_admin.yml]: Do these just no longer exist?

File: config/sync/user.role.content_editor.yml [hunk 11]
```diff
@@ -15,12 +89,8 @@ permissions:
   - 'access administration pages'
   - 'access content overview'
   - 'access files overview'
-  - 'access image_browser entity browser pages'
   - 'access media overview'
-  - 'access media_browser entity browser pages'
-  - 'access shortcuts'
   - 'access user profiles'
-  - 'access vamc_operating_statuses entity browser pages'
```
Comment (swirtSJW) [inline config/sync/user.role.content_editor.yml]: We'll want to test as RyanSt  to see if the operating status page lets him manipulate facility status.

File: config/sync/user.role.content_publisher.yml [hunk 12]
```diff
@@ -161,7 +223,6 @@ permissions:
   - 'view campaign_landing_page revisions'
   - 'view centralized_content revisions'
   - 'view checklist revisions'
-  - 'view cms_help_center in toolbar'
```
Comment (swirtSJW) [inline config/sync/user.role.content_publisher.yml]: Not sure why this is leaving.

File: docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php [hunk 13]
```diff
@@ -28,12 +28,7 @@
  *       "entity:node",
  *       label = @Translation("Node"),
  *       required = FALSE,
- *     ),
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php]: This was throwing an error?

File: tests/phpunit/Security/RolesPermissionsTest.php (hunk 4 above)
Comment (swirtSJW) [reply tests/phpunit/Security/RolesPermissionsTest.php:745]: Thanks for adding this for clarity and helping connect the dots for me.

File: tests/phpunit/Security/RolesPermissionsTest.php (hunk 8 above)
Comment (swirtSJW) [reply tests/phpunit/Security/RolesPermissionsTest.php:768]: I guess not.  It was probably something from days of yore that got dropped but was never removed.

File: config/sync/user.role.authenticated.yml (hunk 9 above)
Comment (ndouglas) [reply config/sync/user.role.authenticated.yml]: Doesn't look like we have a vocabulary named `audience_tags`.  We have `audience_beneficiaries` and `audience_non_beneficiaries` -- maybe `audience_tags` was replaced?

Comment (swirtSJW) [reply config/sync/user.role.authenticated.yml]: explained by link provided farther down

File: config/sync/user.role.content_admin.yml (hunk 10 above)
Comment (swirtSJW) [reply config/sync/user.role.content_admin.yml]: explained by link farther down

Comment (ndouglas) [reply config/sync/user.role.content_admin.yml]: It looks like there's no transition named `stage_for_publishing`; at least, I don't see one in the content moderation workflow.  I don't see any matches for `use moderation sidebar` in the codebase; looks like it's added by the `moderation_sidebar` module, which we don't seem to use.  See [here](https://git.drupalcode.org/project/moderation_sidebar/-/blob/8.x-1.x/moderation_sidebar.permissions.yml).

File: docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php (hunk 13 above)
Comment (ndouglas) [reply docroot/modules/custom/va_gov_workflow_assignments/src/Plugin/Block/EntityMetaDisplay.php]: Yep, there's no ContextDefinition plugin for `entity:node_revision`.  Not sure why the code ever worked TBH.  The block shows up on nodes, the node revisions page, and each node revision page though with the code removed so 🤷‍♂️

File: config/sync/user.role.content_editor.yml (hunk 11 above)
Comment (ndouglas) [reply config/sync/user.role.content_editor.yml]: I:

- logged in
Screen Shot 2022-02-23 at 7 44 34 AM
- went to https://cms-ok9tq0ppwkxwzdj9fdnzmptspq3w8pav.ci.cms.va.gov/node/1010/edit (Pittsburgh)
- edited Beaver County VA clinic
- set the operating status to closed
- saved successfully.

File: docroot/modules/custom/va_gov_db/va_gov_db.install (hunk 1 above)
Comment (ndouglas) [reply docroot/modules/custom/va_gov_db/va_gov_db.install:115]: Should be fine -- it's [the recommended code snippet](https://api.drupal.org/api/drupal/core!modules!taxonomy!taxonomy.module/function/taxonomy_term_load_multiple_by_name/9.3.x).

File: config/sync/user.role.content_editor.yml (hunk 11 above)
Comment (swirtSJW) [reply config/sync/user.role.content_editor.yml:23]: Nice!

**Review by swirtSJW — APPROVED** [review]
Nice work on this @ndouglas     Thanks so much for taking  it on.

**Comment by ndouglas** [comment]
No, @swirtSJW, thank _you_ for reviewing it.  I'll never forgive 💧 for rewriting config and deleting old permissions in the same update 😂 😡

**Pull request merged**